examples/advanced_agents/sandbox_agent/src/nat_sandbox_agent/tools/sandbox/execution.py

```
@@ -0,0 +1,174 @@
+    """
+    logger.info(f"Executing Python code ({len(code)} chars)")
+
+    # Write code to a temp file for better error messages
+    try:
+        await executor.sandbox.write_file(DEFAULT_SCRIPT_PATH, code)
+    except Exception:
+        logger.exception("Failed to write script file")
+        return {
+            "status": "error",
+            "stdout": "",
+            "stderr": executor.truncate(str(e)),
```

Ruff (0.14.13)

Undefined name `e`

(F821)